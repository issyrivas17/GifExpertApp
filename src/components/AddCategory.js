import React,{useState} from 'react'

export const AddCategory = () => {

    const [inputValue,setInputValue] = useState('Hola mundo')

    const handleInputChange= (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => { 
        e.preventDefault(); 
console.log('Submit hecho')
    }

    return (
        <form  onSubmit = {handleSubmit}> 
        <input
       type = "text" 
       value = {inputValue} 
       onChange = {handleInputChange} 
       />
        </form>
      
        
    )
}
