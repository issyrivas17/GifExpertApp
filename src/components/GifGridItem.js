import React from 'react' 
import PropTypes from 'prop-types'

    export const GifGridItem = ({id,title,url}) => {
      /*   console.log(id,title,url);  */
        return (
            <div className= "animate__fadeInBottomRight">
            
            
            
            
    <img src={url} alt = {title} />
    <p> {title}  </p>
            </div>
        )
    }

    GifGridItem.protoTypes = {
      url: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired 
    }