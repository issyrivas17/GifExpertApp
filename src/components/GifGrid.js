    import React from 'react'
    import { useFetchGifs } from '../Hooks/useFetchGifs';
    import {GifGridItem} from '../components/GifGridItem'

        export const GifGrid = ({category}) => {

        const {data:images,loading } =useFetchGifs(category); 

        
            return (
            <>  
                <h3 className="animate__wobble">{category}</h3> 
        
                {loading && <p className="animate__flash"loading></p>}  
                <div className = 'card-grid'>
            
                {
            images.map (img => (
                <GifGridItem 
                key={img.id}
                {... img} />
                ))
            }
                    </div> 
                    </>
                )
        }