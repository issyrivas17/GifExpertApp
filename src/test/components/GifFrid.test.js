import { shallowToJson } from "enzyme-to-json"; 
import {GifGridItem}  from '../../components/GifGridItem'
import React from 'react' 


describe('Pruebas en <GifGridItem/>', () => { 

    const title= 'Un titulo' 
    const url= "https://localhost//algo.jpg" 

    test('Debe de mostrar el componente correctamente', () => { 
       
        const wrapper = shallowToJson(<GifGridItem title ={title} url={url}/>); 
        expect(wrapper).toMatchSnapshot(); 
        
    })
    
    
        

        expect(title)
        
    
    
    
});
