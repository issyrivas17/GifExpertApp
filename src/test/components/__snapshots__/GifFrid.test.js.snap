// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe de mostrar el componente correctamente 1`] = `null`;
